### ISSUE TITLE
`polyglot.value` in internal generic functions

### ISSUE THREAD

**Issue by querenker** [oracle/fastr#122]
I want to extend an internal generic function to handle `polyglot.value` elements. Therefore I added a case for that to the internal generic function `dim`.
```r
> dim.polyglot.value <- function(x) 42
```
Now the dim function called with an element of class `polyglot.value` should return 42. But instead it calls the default dim function, which returns `NULL`:

```r
> l = eval.polyglot('python', '[1, 2, 3]')
> class(l)
[1] "polyglot.value"
> dim(l)
NULL
```

If I try it with an other element and assign the class `polyglot.value` to it, it works like expected:

```r
> x <- 1
> class(x) <- 'polyglot.value'
> dim(x)
[1] 42
```

It also works with a call to `UseMethods`:
```r
> dim2 <- function(x) UseMethod('dim2', x)
> dim2.polyglot.value <- function(x) 42
> dim2(l)
[1] 42
```

**Comment by querenker** [comment]
The problem seem to be, that internal dispatch only occurs on objects and polyglot values are nor treated as objects. I have added an extra issue therefore (#123) and will close this one.

**Comment by steve-s** [comment]
For the record: this will be implemented, see discussion in #123
